crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -3105,35 +3104,46 @@ pub(crate) fn check_ast(
 ///
 /// [`LintContext::report_diagnostic`] can be used to obtain a [`DiagnosticGuard`], which will push
 /// a [`Violation`] to the contained [`OldDiagnostic`] collection on `Drop`.
-pub(crate) struct LintContext<'a> {
+pub(crate) struct LintContext {
     diagnostics: RefCell<Vec<OldDiagnostic>>,
-    source_file: &'a SourceFile,
-    settings: &'a LinterSettings,
+    source_file: SourceFile,
+    rules: RuleTable,
```

Comment:
I found it somewhat convenient that `LintContext` stores the settings, as I think it will allow us to pass `LintContext` in most places where we currently pass `Checker`. 

But I guess it makes sense not to store the settings if we also store them on `Checker`, although I think we should just remove it from `Checker`. Either way. This seems unrelated to this PR and not urgent but maybe something for a contributor issue.